{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import qtlsearch\n",
    "from IPython.display import Image,SVG\n",
    "search = qtlsearch.SEARCH(\n",
    "    \"http://pbg-ld.candygene-nlesc.surf-hosted.nl:8890/sparql\", \n",
    "    \"http://sparql.omabrowser.org/sparql\",\n",
    "    \"https://sparql.uniprot.org/sparql\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    " #tomato\n",
    "tg1 = \"Solyc10g075170.1\"\n",
    "tg2 = \"Solyc10g076240.1\"\n",
    "\n",
    "#tg1 = \"Solyc12g056530.1\"\n",
    "#tg2 = \"Solyc12g056530.1\"\n",
    "\n",
    "tg1 = \"gene72_0-i22\"\n",
    "tg2 = \"gene78_0-i22\"\n",
    "\n",
    "#tomato\n",
    "tg1 = \"Solyc10g075130.1\"\n",
    "tg2 = \"Solyc10g076240.1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Solyc10g075130.1</th>\n",
       "      <td>chromosome 10:58849843-58850038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solyc10g076240.1</th>\n",
       "      <td>chromosome 10:59082990-59084119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         location\n",
       "gene_id                                          \n",
       "Solyc10g075130.1  chromosome 10:58849843-58850038\n",
       "Solyc10g076240.1  chromosome 10:59082990-59084119"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intervalT = search.compute_interval(tg1, tg2)\n",
    "#locationT = search.get_location(\"Solyc12g056530.1\")\n",
    "#d=100000\n",
    "#intervalT = search.make_interval(\n",
    "#    locationT.iloc[0][\"begin_ref\"], \n",
    "#    locationT.iloc[0][\"begin_pos\"]-d, \n",
    "#    locationT.iloc[0][\"end_pos\"]+d)\n",
    "genesT = search.interval_genes(intervalT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "qtls = [genesT.index]\n",
    "go_annotations = search.get_child_annotations(\"GO:0045892\")\n",
    "#print(go_annotations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Search for Solyc10g075140.1\n",
      "- no groups found\n",
      "Search for Solyc10g075150.1\n",
      "- root is http://omabrowser.org/ontology/oma#GROUP_125182\n",
      "- tree of groups fetched: 1\n",
      "- proteins within tree fetched: 2\n",
      "- uniprot proteins within tree fetched: 2\n",
      "  * check proteins (1-2): 0 with required annotation\n",
      "- checked 2 uniprot proteins: 0 with required annotation\n",
      "- checked 2 proteins: 0 linked to uniprot with required annotation\n",
      "Search for Solyc10g075160.1\n",
      "- root is http://omabrowser.org/ontology/oma#GROUP_432941\n",
      "- tree of groups fetched: 121\n",
      "- proteins within tree fetched: 197\n",
      "- uniprot proteins within tree fetched: 162\n"
     ]
    },
    {
     "ename": "HTTPError",
     "evalue": "HTTP Error 414: Request-URI Too Large",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-a2ef9db8089c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     60\u001b[0m                     \u001b[0;31m#check chunks of proteins in uniprot\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m                     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlists_proteins\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 62\u001b[0;31m                         \u001b[0mchecked_proteins_sublist\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msearch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcheck_uniprot_annotations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlists_proteins\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgo_annotations\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     63\u001b[0m                         print(\"  * check proteins (\"+str((i*n)+1)+\"-\"+str(min((i+1)*n,len(tree_proteins)))+\"): \"+\n\u001b[1;32m     64\u001b[0m                               str(len(checked_proteins_sublist))+\" with required annotation\")\n",
      "\u001b[0;32m~/Projecten/Gurnoor/QTLSearch/jupyter/qtlsearch.py\u001b[0m in \u001b[0;36mcheck_uniprot_annotations\u001b[0;34m(self, uniprot_proteins, go_annotations)\u001b[0m\n\u001b[1;32m    251\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msparql_uniprot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetQuery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"proteins\"\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0;34m\"<\"\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\">,<\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0muniprot_proteins\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\">\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"annotations\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m\"<\"\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\">,<\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgo_annotations\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\">\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    252\u001b[0m         \u001b[0;31m# JSON example\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 253\u001b[0;31m         \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msparql_uniprot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconvert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    254\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    255\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"results\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"bindings\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/site-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36mquery\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    796\u001b[0m             \u001b[0;34m@\u001b[0m\u001b[0mrtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mL\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0mQueryResult\u001b[0m\u001b[0;34m}\u001b[0m \u001b[0minstance\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    797\u001b[0m         \"\"\"\n\u001b[0;32m--> 798\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mQueryResult\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_query\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    799\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    800\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mqueryAndConvert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/site-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36m_query\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    776\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mEndPointInternalError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    777\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 778\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    779\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    780\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/site-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36m_query\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    764\u001b[0m                 \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    765\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 766\u001b[0;31m                 \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    767\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturnFormat\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    768\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0murllib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mHTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    220\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    221\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 222\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    223\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    224\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    529\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mprocessor\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprocess_response\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprotocol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    530\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprocessor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 531\u001b[0;31m             \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    532\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    533\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36mhttp_response\u001b[0;34m(self, request, response)\u001b[0m\n\u001b[1;32m    639\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;36m200\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0mcode\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m300\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    640\u001b[0m             response = self.parent.error(\n\u001b[0;32m--> 641\u001b[0;31m                 'http', request, response, code, msg, hdrs)\n\u001b[0m\u001b[1;32m    642\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    643\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36merror\u001b[0;34m(self, proto, *args)\u001b[0m\n\u001b[1;32m    567\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhttp_err\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    568\u001b[0m             \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'default'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'http_error_default'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0morig_args\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 569\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_chain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    570\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    571\u001b[0m \u001b[0;31m# XXX probably also want an abstract factory that knows when it makes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    501\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    502\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 503\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    504\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    505\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/miniconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36mhttp_error_default\u001b[0;34m(self, req, fp, code, msg, hdrs)\u001b[0m\n\u001b[1;32m    647\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPDefaultErrorHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    648\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mhttp_error_default\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 649\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    650\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    651\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPRedirectHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHTTPError\u001b[0m: HTTP Error 414: Request-URI Too Large"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "p_uniprot_reviewed = 1.00\n",
    "p_uniprot_unreviewed = 0.95\n",
    "loss_up_ortholog = 0.85\n",
    "loss_down_ortholog = 0.85\n",
    "loss_up_paralog = 0.7225\n",
    "loss_down_paralog = 0.7225\n",
    "\n",
    "#go_annotations = search.get_child_annotations(\"GO:0010154\")\n",
    "\n",
    "hog_group_trees = pd.Series()\n",
    "hog_group_genes = pd.Series()\n",
    "#root in hog tree for each gene\n",
    "qtl_gene_roots = pd.Series()\n",
    "\n",
    "qtl_gene_protein = pd.Series()\n",
    "gene_p_initial = pd.Series()\n",
    "\n",
    "\n",
    "for qtl in qtls:\n",
    "    for gene in qtl:\n",
    "        if not(gene in qtl_gene_protein.keys()):\n",
    "            p_qtl_initial = 1.0/len(qtl)        \n",
    "            #start searching\n",
    "            print(\"Search for \"+gene)\n",
    "            #first, go up in the hog-tree\n",
    "            parent_groups = search.get_parent_groups(gene)\n",
    "            if len(parent_groups)>0:\n",
    "                #define the root of the tree\n",
    "                qtl_gene_roots[gene] = parent_groups.index[0]\n",
    "                qtl_gene_protein[gene] = parent_groups.loc[parent_groups.index[0]].protein\n",
    "                print(\"- root is \"+qtl_gene_roots[gene])\n",
    "                if qtl_gene_roots[gene] in hog_group_trees.index:\n",
    "                    print(\"- tree already created\")\n",
    "                else:    \n",
    "                    #go down the hog-tree, just to find tree structure\n",
    "                    hog_group_trees[qtl_gene_roots[gene]] = search.get_child_groups(qtl_gene_roots[gene])\n",
    "                    hog_group_trees[qtl_gene_roots[gene]].loc[:,\"p_initial\"] = pd.Series(0.0, index=hog_group_trees[qtl_gene_roots[gene]].index)        \n",
    "                    hog_group_trees[qtl_gene_roots[gene]].loc[:,\"p_up\"] = pd.Series(0.0, index=hog_group_trees[qtl_gene_roots[gene]].index)        \n",
    "                    hog_group_trees[qtl_gene_roots[gene]].loc[:,\"p_down\"] = pd.Series(0.0, index=hog_group_trees[qtl_gene_roots[gene]].index)        \n",
    "                    print(\"- tree of groups fetched: \"+str(len(hog_group_trees[qtl_gene_roots[gene]])))\n",
    "                    #go down again, now to find proteins\n",
    "                    tree_proteins = search.get_child_proteins(qtl_gene_roots[gene])\n",
    "                    tree_proteins_uniprot = search.get_child_proteins_uniprot(qtl_gene_roots[gene])\n",
    "                    print(\"- proteins within tree fetched: \"+str(len(tree_proteins)))\n",
    "                    print(\"- uniprot proteins within tree fetched: \"+str(len(tree_proteins_uniprot)))\n",
    "                    #create final list of checked proteins\n",
    "                    hog_group_genes[qtl_gene_roots[gene]] = tree_proteins\n",
    "                    hog_group_genes[qtl_gene_roots[gene]].loc[:,\"reviewed\"] = pd.Series(\"unknown\", index=hog_group_genes[qtl_gene_roots[gene]].index)\n",
    "                    hog_group_genes[qtl_gene_roots[gene]].loc[:,\"p_initial\"] = pd.Series(0.0, index=hog_group_genes[qtl_gene_roots[gene]].index)\n",
    "                    hog_group_genes[qtl_gene_roots[gene]].loc[:,\"p_up\"] = pd.Series(0.0, index=hog_group_genes[qtl_gene_roots[gene]].index)\n",
    "                    hog_group_genes[qtl_gene_roots[gene]].loc[:,\"p_final\"] = pd.Series(0.0, index=hog_group_genes[qtl_gene_roots[gene]].index)\n",
    "                    #create uniprot list for checking\n",
    "                    list_proteins = list(tree_proteins_uniprot.index)\n",
    "                    #divide proteins into chunks (sparql endpoint doesn't allow huge lists)\n",
    "                    n=50\n",
    "                    lists_proteins = [list_proteins[i * n:(i + 1) * n] for i in range((len(list_proteins) + n - 1) // n )]          \n",
    "                    checked_proteins_list = []\n",
    "                    #check chunks of proteins in uniprot\n",
    "                    for i in range(0,len(lists_proteins)):\n",
    "                        checked_proteins_sublist = search.check_uniprot_annotations(lists_proteins[i], list(go_annotations.index))\n",
    "                        print(\"  * check proteins (\"+str((i*n)+1)+\"-\"+str(min((i+1)*n,len(tree_proteins)))+\"): \"+\n",
    "                              str(len(checked_proteins_sublist))+\" with required annotation\")\n",
    "                        #collect group labels to give some indication of origin\n",
    "                        mask = list(tree_proteins_uniprot.loc[checked_proteins_sublist.index].protein.unique())\n",
    "                        sublist_labels = list(tree_proteins.loc[mask].label.dropna().unique())\n",
    "                        if len(sublist_labels)>0:\n",
    "                              print(\"    -> \"+\", \".join(sublist_labels))\n",
    "                        #add checked proteins from chunk to the collective list        \n",
    "                        checked_proteins_list.append(checked_proteins_sublist)\n",
    "                    checked_proteins = pd.concat(checked_proteins_list)                 \n",
    "                    #update reviewed\n",
    "                    hog_group_genes[qtl_gene_roots[gene]].reviewed = None\n",
    "                    if qtl_gene_protein[gene] in gene_p_initial.index:\n",
    "                        hog_group_genes[qtl_gene_roots[gene]].loc[qtl_gene_protein[gene],\"p_initial\"] = max(gene_p_initial[qtl_gene_protein[gene]], p_qtl_initial)\n",
    "                    else:\n",
    "                        hog_group_genes[qtl_gene_roots[gene]].loc[qtl_gene_protein[gene],\"p_initial\"] = p_qtl_initial\n",
    "                    if len(checked_proteins)>0:\n",
    "                        maskUnreviewed = hog_group_genes[qtl_gene_roots[gene]].loc[tree_proteins_uniprot.loc[checked_proteins.loc[checked_proteins.reviewed==\"false\"].index].protein.unique()]\n",
    "                        maskReviewed = hog_group_genes[qtl_gene_roots[gene]].loc[tree_proteins_uniprot.loc[checked_proteins.loc[checked_proteins.reviewed==\"true\"].index].protein.unique()]\n",
    "                        hog_group_genes[qtl_gene_roots[gene]].loc[maskUnreviewed.index,\"reviewed\"] = False\n",
    "                        hog_group_genes[qtl_gene_roots[gene]].loc[maskReviewed.index,\"reviewed\"] = True\n",
    "                        #update probability                        \n",
    "                        for reviewedProtein in hog_group_genes[qtl_gene_roots[gene]][hog_group_genes[qtl_gene_roots[gene]].reviewed==True].index:\n",
    "                            hog_group_genes[qtl_gene_roots[gene]].loc[reviewedProtein,\"p_initial\"] = max(hog_group_genes[qtl_gene_roots[gene]].loc[reviewedProtein,\"p_initial\"],p_uniprot_reviewed)\n",
    "                        for unreviewedProtein in hog_group_genes[qtl_gene_roots[gene]][hog_group_genes[qtl_gene_roots[gene]].reviewed==False].index:\n",
    "                            hog_group_genes[qtl_gene_roots[gene]].loc[unreviewedProtein,\"p_initial\"] = max(hog_group_genes[qtl_gene_roots[gene]].loc[unreviewedProtein,\"p_initial\"],p_uniprot_unreviewed)\n",
    "                    for positiveProbablilityProtein in hog_group_genes[qtl_gene_roots[gene]].loc[hog_group_genes[qtl_gene_roots[gene]][\"p_initial\"]>0].index:\n",
    "                        if positiveProbablilityProtein in gene_p_initial.index:\n",
    "                            gene_p_initial[positiveProbablilityProtein] = max(hog_group_genes[qtl_gene_roots[gene]].loc[positiveProbablilityProtein,\"p_initial\"],gene_p_initial[positiveProbablilityProtein])\n",
    "                        else:        \n",
    "                            gene_p_initial[positiveProbablilityProtein] = hog_group_genes[qtl_gene_roots[gene]].loc[positiveProbablilityProtein,\"p_initial\"]\n",
    "                    #report results\n",
    "                    print(\"- checked \"+str(len(list_proteins))+\" uniprot proteins: \"+str(len(checked_proteins))+\" with required annotation\")\n",
    "                    #including details about reviewed status\n",
    "                    if len(checked_proteins)>0:\n",
    "                        reviewedList = checked_proteins.groupby(\"reviewed\")\n",
    "                        if len(reviewedList)>0:\n",
    "                            for label,number in reviewedList.size().items():\n",
    "                                print(\"  -> for \"+str(number)+\" of these \"+str(len(list_proteins))+\" items, reviewed is \"+str(label))                     \n",
    "                    print(\"- checked \"+str(len(tree_proteins))+\" proteins: \"+str(len(hog_group_genes[qtl_gene_roots[gene]][\"reviewed\"].dropna()))+\" linked to uniprot with required annotation\")\n",
    "                    if len(hog_group_genes[qtl_gene_roots[gene]])>0:\n",
    "                        reviewedList = hog_group_genes[qtl_gene_roots[gene]].groupby(\"reviewed\")\n",
    "                        if len(reviewedList)>0:\n",
    "                            for label,number in reviewedList.size().items():\n",
    "                                print(\"  -> for \"+str(number)+\" of these \"+str(len(hog_group_genes[qtl_gene_roots[gene]]))+\" items reviewed is marked as \"+str(label))                     \n",
    "\n",
    "            else:\n",
    "                #no hog groups, can't do anything with this...\n",
    "                print(\"- no groups found\")\n",
    "                \n",
    "#update initial probabilities from other genes and qtls    \n",
    "for gene in qtl_gene_protein.keys():\n",
    "    for protein in hog_group_genes[qtl_gene_roots[gene]].keys():\n",
    "        if protein in gene_p_initial.keys():\n",
    "            hog_group_genes[qtl_gene_roots[gene]].loc[protein,\"p_initial\"] = gene_p_initial[protein];\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skip Solyc10g075140.1\n",
      "Compute for Solyc10g075150.1\n",
      "Compute for Solyc10g075160.1\n",
      "Compute for Solyc10g075170.1\n",
      "Skip Solyc10g076170.1\n",
      "Compute for Solyc10g076180.1\n",
      "Compute for Solyc10g076190.1\n",
      "Compute for Solyc10g076200.1\n",
      "Compute for Solyc10g076210.1\n",
      "Compute for Solyc10g076220.1\n",
      "Compute for Solyc10g076230.1\n"
     ]
    }
   ],
   "source": [
    "def get_p_up(group):\n",
    "    children = hog_group_trees[qtl_gene_roots[gene]].loc[hog_group_trees[qtl_gene_roots[gene]].parent==group]\n",
    "    proteins = hog_group_genes[qtl_gene_roots[gene]].loc[hog_group_genes[qtl_gene_roots[gene]].group==group]\n",
    "    type = hog_group_trees[qtl_gene_roots[gene]].loc[group,\"type\"]\n",
    "    p = hog_group_trees[qtl_gene_roots[gene]].loc[group,\"p_initial\"]\n",
    "    for protein in proteins.index:\n",
    "        if type==\"ortholog\":\n",
    "            p +=  loss_up_ortholog * hog_group_genes[qtl_gene_roots[gene]].loc[protein,\"p_initial\"]\n",
    "        elif type==\"paralog\":    \n",
    "            p +=  loss_up_paralog * hog_group_genes[qtl_gene_roots[gene]].loc[protein,\"p_initial\"]\n",
    "    for child in children.index:\n",
    "        if type==\"ortholog\":\n",
    "            p +=  loss_up_ortholog * get_p_up(child)\n",
    "        elif type==\"paralog\":    \n",
    "            p +=  loss_up_paralog * get_p_up(child)   \n",
    "    hog_group_trees[qtl_gene_roots[gene]].loc[group,\"p_up\"] = p\n",
    "    #print(\"Group \"+group+\" - \"+type+\": \"+str(p))\n",
    "    return p;\n",
    "\n",
    "def set_p_down(group):\n",
    "    children = hog_group_trees[qtl_gene_roots[gene]].loc[hog_group_trees[qtl_gene_roots[gene]].parent==group]\n",
    "    proteins = hog_group_genes[qtl_gene_roots[gene]].loc[hog_group_genes[qtl_gene_roots[gene]].group==group]\n",
    "    type = hog_group_trees[qtl_gene_roots[gene]].loc[group,\"type\"]\n",
    "    p = hog_group_trees[qtl_gene_roots[gene]].loc[group,\"p_up\"]\n",
    "    parent = hog_group_trees[qtl_gene_roots[gene]].loc[qtl_gene_roots[gene],\"parent\"]\n",
    "    if not(parent==None):\n",
    "      parent_type = hog_group_trees[qtl_gene_roots[gene]].loc[group,\"type\"]  \n",
    "      parent_p = hog_group_trees[qtl_gene_roots[gene]].loc[parent,\"p_down\"]\n",
    "      if parent_type==\"ortholog\":  \n",
    "          p = max(p,loss_down_ortholog*parent_p)\n",
    "      elif parent_type==\"paralog\":\n",
    "          p = max(p,loss_down_paralog*parent_p)\n",
    "    hog_group_trees[qtl_gene_roots[gene]].loc[group,\"p_down\"] = p\n",
    "    for protein in proteins.index:  \n",
    "        p_protein = hog_group_genes[qtl_gene_roots[gene]].loc[protein,\"p_initial\"]\n",
    "        if type==\"ortholog\":\n",
    "            p_protein = max(loss_down_ortholog*p,p_protein)\n",
    "        elif type==\"paralog\":\n",
    "            p_protein = max(loss_down_paralog*p,p_protein)    \n",
    "        hog_group_genes[qtl_gene_roots[gene]].loc[protein,\"p_final\"] = p_protein\n",
    "    for child in children.index:        \n",
    "        set_p_down(child)\n",
    "    \n",
    "for qtl in qtls:\n",
    "    for gene in qtl:        \n",
    "        if gene in qtl_gene_roots.keys():\n",
    "            print(\"Compute for \"+gene)\n",
    "            #reset\n",
    "            hog_group_trees[qtl_gene_roots[gene]].loc[:,\"p_up\"] = 0.0\n",
    "            hog_group_trees[qtl_gene_roots[gene]].loc[:,\"p_down\"] = 0.0\n",
    "            hog_group_genes[qtl_gene_roots[gene]].loc[:,\"p_final\"] = 0.0\n",
    "            #go up\n",
    "            get_p_up(qtl_gene_roots[gene])\n",
    "            #go down\n",
    "            set_p_down(qtl_gene_roots[gene])\n",
    "        else:\n",
    "            print(\"Skip \"+gene)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== START QTL 1 ===\n",
      "Solyc10g075140.1\n",
      "Solyc10g075150.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05946\n",
      "Solyc10g075160.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05947\n",
      "Solyc10g075170.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05948\n",
      "Solyc10g076170.1\n",
      "Solyc10g076180.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05950\n",
      "Solyc10g076190.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05951\n",
      "Solyc10g076200.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05952\n",
      "Solyc10g076210.1\t0.0\t0.04745511363636364\thttps://omabrowser.org/oma/info/SOLLC05953\n",
      "Solyc10g076220.1\t0.0\t0.04745511363636364\thttps://omabrowser.org/oma/info/SOLLC05954\n",
      "Solyc10g076230.1\t0.09090909090909091\t0.09090909090909091\thttps://omabrowser.org/oma/info/SOLLC05955\n",
      "=== END QTL 1 ===\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#report\n",
    "for i in range(0,len(qtls)):\n",
    "    print(\"=== START QTL \"+str(i+1)+\" ===\")\n",
    "    for gene in qtls[i]:\n",
    "        if gene in qtl_gene_roots.keys():\n",
    "            print(gene\n",
    "                  +\"\\t\"\n",
    "                  +str(hog_group_genes[qtl_gene_roots[gene]].loc[qtl_gene_protein[gene],\"p_initial\"])\n",
    "                  +\"\\t\"\n",
    "                  +str(hog_group_genes[qtl_gene_roots[gene]].loc[qtl_gene_protein[gene],\"p_final\"])\n",
    "                  +\"\\t\"\n",
    "                  +qtl_gene_protein[gene]\n",
    "                 ) \n",
    "        else:\n",
    "            print(gene)    \n",
    "    print(\"=== END QTL \"+str(i+1)+\" ===\")\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://omabrowser.org/oma/info/SOLLC05950'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qtl_gene_protein[\"Solyc10g076180.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>label</th>\n",
       "      <th>reviewed</th>\n",
       "      <th>p_initial</th>\n",
       "      <th>p_up</th>\n",
       "      <th>p_final</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/RHIOR15394</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Fungi</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/CREFR03759</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Ichthyosporea</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/AMPQE22899</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD07600</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD03443</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD03337</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN25118</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/AMPQE19500</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD00289</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD03886</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/AMPQE00483</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD06487</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/THEKT12588</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD06001</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/AMPQE20676</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD06130</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/TRIAD08769</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/NEMVE10394</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/HYDVU09570</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/NEMVE11245</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/HYDVU15441</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/MEGSC05819</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Cyclorrhapha</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/SALR501037</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Salpingoecidae</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/MONBE05483</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Salpingoecidae</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/SALR510287</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Salpingoecidae</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/MONBE08485</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Salpingoecidae</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/ECHTE07569</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Amniota</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/PELSI02576</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Amniota</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/MONDO09689</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Amniota</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/PELSI03562</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#GROUP_43447...</td>\n",
       "      <td>Amniota</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN03314</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN03350</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN04150</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN04413</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN04964</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN05521</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN05555</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN06249</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN06404</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN06595</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN06607</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN07340</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN07490</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN07854</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN08102</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN08845</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN09162</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN09564</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN10794</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN11177</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN11655</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN11785</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN11837</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN11913</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN12226</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LINUN12228</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LOTGI02620</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LOTGI02754</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/LOTGI02813</th>\n",
       "      <td>http://omabrowser.org/ontology/oma#PARALOG_GRO...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.495906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>https://omabrowser.org/oma/info/SOLLC11031</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3001 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                        group  \\\n",
       "protein                                                                                         \n",
       "https://omabrowser.org/oma/info/RHIOR15394  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/CREFR03759  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/AMPQE22899  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD07600  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD03443  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD03337  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/LINUN25118  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/AMPQE19500  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD00289  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD03886  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/AMPQE00483  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD06487  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/THEKT12588  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD06001  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/AMPQE20676  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD06130  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/TRIAD08769  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/NEMVE10394  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/HYDVU09570  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/NEMVE11245  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/HYDVU15441  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/MEGSC05819  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/SALR501037  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/MONBE05483  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/SALR510287  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/MONBE08485  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/ECHTE07569  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/PELSI02576  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/MONDO09689  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "https://omabrowser.org/oma/info/PELSI03562  http://omabrowser.org/ontology/oma#GROUP_43447...   \n",
       "...                                                                                       ...   \n",
       "https://omabrowser.org/oma/info/LINUN03314  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN03350  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN04150  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN04413  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN04964  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN05521  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN05555  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN06249  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN06404  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN06595  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN06607  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN07340  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN07490  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN07854  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN08102  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN08845  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN09162  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN09564  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN10794  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN11177  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN11655  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN11785  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN11837  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN11913  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN12226  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LINUN12228  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LOTGI02620  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LOTGI02754  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/LOTGI02813  http://omabrowser.org/ontology/oma#PARALOG_GRO...   \n",
       "https://omabrowser.org/oma/info/SOLLC11031                                                NaN   \n",
       "\n",
       "                                                     label reviewed  \\\n",
       "protein                                                               \n",
       "https://omabrowser.org/oma/info/RHIOR15394           Fungi     None   \n",
       "https://omabrowser.org/oma/info/CREFR03759   Ichthyosporea     None   \n",
       "https://omabrowser.org/oma/info/AMPQE22899         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD07600         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD03443         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD03337         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/LINUN25118         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/AMPQE19500         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD00289         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD03886         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/AMPQE00483         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD06487         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/THEKT12588         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD06001         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/AMPQE20676         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD06130         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/TRIAD08769         Metazoa     None   \n",
       "https://omabrowser.org/oma/info/NEMVE10394        Cnidaria     None   \n",
       "https://omabrowser.org/oma/info/HYDVU09570        Cnidaria     None   \n",
       "https://omabrowser.org/oma/info/NEMVE11245        Cnidaria     None   \n",
       "https://omabrowser.org/oma/info/HYDVU15441        Cnidaria     None   \n",
       "https://omabrowser.org/oma/info/MEGSC05819    Cyclorrhapha     None   \n",
       "https://omabrowser.org/oma/info/SALR501037  Salpingoecidae     None   \n",
       "https://omabrowser.org/oma/info/MONBE05483  Salpingoecidae     None   \n",
       "https://omabrowser.org/oma/info/SALR510287  Salpingoecidae     None   \n",
       "https://omabrowser.org/oma/info/MONBE08485  Salpingoecidae     None   \n",
       "https://omabrowser.org/oma/info/ECHTE07569         Amniota     None   \n",
       "https://omabrowser.org/oma/info/PELSI02576         Amniota     None   \n",
       "https://omabrowser.org/oma/info/MONDO09689         Amniota     None   \n",
       "https://omabrowser.org/oma/info/PELSI03562         Amniota     None   \n",
       "...                                                    ...      ...   \n",
       "https://omabrowser.org/oma/info/LINUN03314            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN03350            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN04150            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN04413            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN04964            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN05521            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN05555            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN06249            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN06404            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN06595            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN06607            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN07340            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN07490            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN07854            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN08102            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN08845            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN09162            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN09564            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN10794            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN11177            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN11655            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN11785            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN11837            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN11913            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN12226            None     None   \n",
       "https://omabrowser.org/oma/info/LINUN12228            None     None   \n",
       "https://omabrowser.org/oma/info/LOTGI02620            None     None   \n",
       "https://omabrowser.org/oma/info/LOTGI02754            None     None   \n",
       "https://omabrowser.org/oma/info/LOTGI02813            None     None   \n",
       "https://omabrowser.org/oma/info/SOLLC11031             NaN      NaN   \n",
       "\n",
       "                                            p_initial  p_up   p_final  \n",
       "protein                                                                \n",
       "https://omabrowser.org/oma/info/RHIOR15394     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/CREFR03759     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/AMPQE22899     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD07600     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD03443     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD03337     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/LINUN25118     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/AMPQE19500     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD00289     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD03886     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/AMPQE00483     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD06487     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/THEKT12588     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD06001     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/AMPQE20676     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD06130     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/TRIAD08769     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/NEMVE10394     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/HYDVU09570     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/NEMVE11245     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/HYDVU15441     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/MEGSC05819     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/SALR501037     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/MONBE05483     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/SALR510287     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/MONBE08485     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/ECHTE07569     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/PELSI02576     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/MONDO09689     0.0000   0.0  0.000000  \n",
       "https://omabrowser.org/oma/info/PELSI03562     0.0000   0.0  0.000000  \n",
       "...                                               ...   ...       ...  \n",
       "https://omabrowser.org/oma/info/LINUN03314     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN03350     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN04150     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN04413     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN04964     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN05521     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN05555     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN06249     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN06404     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN06595     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN06607     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN07340     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN07490     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN07854     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN08102     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN08845     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN09162     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN09564     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN10794     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN11177     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN11655     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN11785     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN11837     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN11913     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN12226     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LINUN12228     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LOTGI02620     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LOTGI02754     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/LOTGI02813     0.0000   0.0  0.495906  \n",
       "https://omabrowser.org/oma/info/SOLLC11031     0.0625   NaN  0.000000  \n",
       "\n",
       "[3001 rows x 6 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hog_group_genes[qtl_gene_roots[\"Solyc12g056490.1\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(checked_proteins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
